packages/world/src/modules/keysintable/KeysInTableHook.sol

```diff
@@ -45,19 +58,47 @@ contract KeysInTableHook is IStoreHook {
       // Delete the index as the key is not in the table
       UsedKeysIndex.deleteRecord(tableId, keysHash);
 
-      uint32 length = KeysInTable.getLength(tableId);
+      uint256 length = KeysInTable.lengthKeys0(tableId);
 
       if (length == 1) {
         // Delete the list of keys in this table
         KeysInTable.deleteRecord(tableId);
       } else {
-        bytes32 lastKey = KeysInTable.getKeys(tableId)[length - 1];
+        if (key.length > 0) {
```

Comment:
I wish we could avoid this switch statement, but we can't index the key fields programmatically

Reply:
technically we could just use the lower level `StoreCore` methods directly (and programatically provide the `fieldIndex` param), but we'd lose the type helpers so probably not worth it